trie/hasher.go

```diff
@@ -149,8 +149,8 @@ func (h *hasher) hashFullNodeChildren(n *fullNode) (collapsed *fullNode, cached
 }
 
 // shortnodeToHash creates a hashNode from a shortNode. The supplied shortnode
-// should have hex-type Key, which will be converted (without modification)
-// into compact form for RLP encoding.
+// should have compact-type Key, which has been converted (without modification)
+// from hex form for RLP encoding.
```

Comment:
I _think_ the comment is correct as is, and becomes incorrect with this PR. 

As it was, it said: 
- The `n` should have the key on expanded hex form. When writing the new hashNode, the key is compacted (which it needs to be for the hashing to be correct), but the key in `n` is left unmodified. 

As it is now, it says that `n` needs to have the key on compact form, and the api doc says it "has been converted" -- which is a strange way to phrase it. What does it mean that something "has" been done? Does it mean that the caller needs to do it before invoking the method? Does it mean that at the time of writing, all known callers have converted it?

Reply:
Yeah, what I saw in relevant code was that the key in `n` has been compacted before calling `shortnodeToHash`  just as the following (trie/hasher.go:77)
![image](https://user-images.githubusercontent.com/8242525/125945438-86211a5e-2cfa-4c38-9f5e-0c245ffb97bb.png)
Is `collapsed` the shortNode whose key was compacted already? That's what I felt confused.